lesson_02/main.py

```diff
@@ -0,0 +1,77 @@
+#script on google cloud function to push clouud storage
+
+import requests
+from google.cloud import storage
+import requests
+
+url = 'https://fake-api-vycpfa6oca-uc.a.run.app/sales?date=2022-08-09' #&page=2
+headers={'Authorization': '2b8d97ce57d401abd89f45b0079d8790edd940e6'}
+
+bucket_name = 'test_de_course'
+
+storage_client = storage.Client()
+
+def upload_blob(bucket, data, destination_blob_name):
+    blob = bucket.blob(destination_blob_name)
+    data = str(data)
+    blob.upload_from_string(data, content_type='application/json')
+
+
+def get_request_to_api(url, headers, params):
+    data = requests.get(url, headers=headers, params=params)
+    return data.json()
+
+def main (request):
+    files_created = []
+
+    try:
+        request_json = request.get_json()
+        #request_json = request
+        date  = request_json.get('date') 
+        raw_dir = request_json.get('raw_dir')
+        del request_json
+
+        if date is None:
```

Comment:
What if the data is not non, but malformatted? For example, 05-2022-05?
It's better to use a library for validating input values.